File: Sources/Sentry/SentryUserFeedback.m
#import "SentryUserFeedback.h"
#import "SentrySwift.h"
#import <Foundation/Foundation.h>

typedef enum : NSUInteger {
    /** A user feedback attached to a transaction or error event. */
    kSentryUserFeedbackTypeAttached,

    /** A user feedback sent as its own event independent of any other event. */
    kSentryUserFeedbackTypeStandalone,
} SentryUserFeedbackType;

@implementation SentryUserFeedback {
    SentryUserFeedbackType _type;
}

- (instancetype)init
{
    if (self = [super init]) {
        _type = kSentryUserFeedbackTypeStandalone;
        _eventId = [[SentryId alloc] init];
    }
    return self;
}

- (instancetype)initWithEventId:(SentryId *)eventId
{
    if (self = [super init]) {
        _eventId = eventId;
        _email = @"";
        _name = @"";
        _comments = @"";
        _type = kSentryUserFeedbackTypeAttached;
    }
    return self;
}

- (NSDictionary<NSString *, id> *)serialize
{
    switch (_type) {
    case kSentryUserFeedbackTypeAttached:
        return @{
            @"event_id" : self.eventId.sentryIdString,
            @"email" : self.email,
            @"name" : self.name,
            @"comments" : self.comments
        };
    case kSentryUserFeedbackTypeStandalone:
        return @{

        };
    }
}

@end

File: Sources/Swift/Integrations/UserFeedback/SentryUserFeedback.swift
import Foundation

/// Adds additional information about what happened to an event.
/// @deprecated Use `SentryFeedback`.
@objc(SentryUserFeedback)
@available(*, deprecated, message: "Use SentryFeedback.")
public class SentryUserFeedback: NSObject, SentrySerializable {
    
    /// The eventId of the event to which the user feedback is associated.
    @objc public private(set) var eventId: SentryId
    
    /// The name of the user.
    @objc public var name: String
    
    /// The email of the user.
    @objc public var email: String
    
    /// Comments of the user about what happened.
    @objc public var comments: String
    
    /// Initializes SentryUserFeedback and sets the required eventId.
    /// - Parameter eventId: The eventId of the event to which the user feedback is associated.
    @objc public init(eventId: SentryId) {
        self.eventId = eventId
        self.email = ""
        self.name = ""
        self.comments = ""
        super.init()
    }
    
    public func serialize() -> [String: Any] {
        return [
            "event_id": eventId.sentryIdString,
            "email": email,
            "name": name,
            "comments": comments
        ]
    }
} 
